stock_picking_purchase_state/models/purchase_order.py

```diff
@@ -0,0 +1,32 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (http://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import api, fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    picking_status = fields.Selection([
+        ('draft', 'Draft'),
+        ('waiting', 'Waiting Another Operation'),
+        ('confirmed', 'Waiting'),
+        ('assigned', 'Ready'),
+        ('done', 'Done'),
+        ('cancel', 'Cancelled'),
+    ], string='Picking Status', compute='_compute_picking_status',
+       readonly=True, store=True,
+       help=" * Draft: The transfer is not confirmed yet. Reservation doesn't apply.\n"
+            " * Waiting another operation: This transfer is waiting for another operation before being ready.\n"
+            " * Waiting: The transfer is waiting for the availability of some products.\n(a) The shipping policy is \"As soon as possible\": no product could be reserved.\n(b) The shipping policy is \"When all products are ready\": not all the products could be reserved.\n"
+            " * Ready: The transfer is ready to be processed.\n(a) The shipping policy is \"As soon as possible\": at least one product has been reserved.\n(b) The shipping policy is \"When all products are ready\": all product have been reserved.\n"
+            " * Done: The transfer has been processed.\n"
+            " * Cancelled: The transfer has been cancelled.")
+
+    @api.depends('picking_ids.state')
+    def _compute_picking_status(self):
+        for order in self:
+            if order.picking_ids.filtered(lambda p: p.state != "cancel"):
+                order.picking_status = order.picking_ids.filtered(lambda p: p.state != "cancel").sorted(key="scheduled_date")[-1].state
+            else:
+                order.picking_status = False
```

Comment:
Proposed refactoring in order to reduce line character count:
```suggestion
            pickings = order.picking_ids.filtered(
                lambda p: p.state != "cancel"
            )
            if pickings:
                order.picking_status = pickings.sorted(
                    key="scheduled_date"
                )[-1].state
            else:
                order.picking_status = False
```